wire.md

```diff
@@ -0,0 +1,1008 @@
+lacks sufficient resources at the time to return thousands or hundreds of
+thousands of chat message hashes.
+
+A `limit` of 0 MUST be understood as having no maximum on the number of hashes
+the requester wishes to receive.
+
+##### 6.3.2.5 Channel State Request
+
+Request posts that describe the current state of a channel and its members, and
+optionally subscribe to future state changes.
+
+field          | type               | desc
+---------------|--------------------|-----------------------------------
+`channel_len`  | `varint`           | length of the channel's name, in bytes 
+`channel`      | `u8[channel_len] ` | channel name (UTF-8)
+`future`       | `varint`           | whether to include live / future state hashes
+
+`msg_type` MUST be set to `5`.
+
+A responder receiving this request MUST respond with 1 or more Hash Response
+messages, with only posts that relate to the current state of the channel.
+Requesters MAY discard hashes mapping to posts that do not contain relevant
+information.
+
+See Section 5.4.4 for context on what comprises channel state. Chat messages
+MUST NOT be included in responses to this request. Clients MUST be able to
+handle the hashes of unexpected types appearing in responses, and MAY choose
+for themselves whether to discard them or not.
+
+`future` MUST be set to either `1` or `0`.
+
+If `future = 1`, the responder SHOULD respond with future channel state changes
+as they become known to the responder, and the request SHOULD be held open
+indefinitely on both the requester and responder side until a Cancel Request is
+issued by the requester, or the responder elects to end the request by sending
+a Hash Response with `hash_count = 0`.
+
+If `future = 1` and a post that is part of the latest state for a channel is
+deleted, the responder MUST immediately send the hash of the next-latest
+piece of state of that same type as a Hash Response. For example, if the latest
+`post/topic` setting the channel's topic string is deleted by its author with a
+`post/delete` post, the hash of the second-latest `post/topic` for that channel
+SHOULD be sent.
+
+If `future = 0`, only the latest state posts will be included, and the request
+MUST NOT be held open.
+
+##### 6.3.2.6 Channel List Request
+
+Request a list of known channels from peers.
+
+field          | type               | desc
+---------------|--------------------|-----------------------------------
+`offset`       | `varint`           | number of channel names to skip (`0` to skip none)
+`limit`        | `varint`           | maximum number of channel names to return
+
+`msg_type` MUST be set to `6`.
+
+This request returns zero or more Channel List Response messages.
+
+If `limit` is 0, the responder MUST respond with all known channels (after
+skipping the first `offset` entries).
+
+The `offset` field can be combined with the `limit` field to allow clients to
+paginate through the list of all channel names known by a peer.
```

Comment:
To paginate, they must be consistently sorted. The spec should require this. It might even require an ordering to make pagination consistent irrespective of the peer.

Reply:
I mention the stable sort requirement in the Channel List Response message, where it's used. I'll think about whether it makes sense to specify a sort method.

Reply:
I thought about it, and agree that a sort should be specified.

> The channel names SHOULD be sorted lexicographically in ascending order, so that requesters can effectively sent subsequent requests to paginate through the results.